parsers/ES.py

```diff
@@ -113,17 +159,61 @@ def check_valid_parameters(
 
 
 def fetch_island_data(
-    zone_key: ZoneKey, session: Session, target_datetime: Optional[datetime]
-) -> List[Response]:
+    zone_key: ZoneKey,
+    session: Session,
+    target_datetime: Optional[datetime],
+    data_mapping: Dict = DATA_MAPPING,
+):
     """Fetch data for the given zone key."""
+    tz = TIMEZONES_MAPPING[zone_key]
     if target_datetime is None:
-        date = target_datetime
+        date = utcnow().to(tz).format("YYYY-MM-DD")
     else:
         date = target_datetime.strftime("%Y-%m-%d")
-
-    data: List[Response] = ZONE_FUNCTION_MAP[zone_key](session).get_all(date)
-    if data:
-        return data
+    system = ZONE_MAPPING[zone_key]
+    zone = API_CODE_MAPPING[zone_key]
+    res = session.get(
+        f"https://demanda.ree.es/WSvisionaMoviles{system}Rest/resources/demandaGeneracion{system}?curva={zone}&fecha={date}"
+    )
+    if not res.ok:
+        raise ParserException(
+            "ES.py",
+            f"Failed fetching data for {zone_key}",
+            zone_key,
+        )
+    json = loads(res.text.replace("null(", "", 1).replace(r");", "", 1))
+    data = json["valoresHorariosGeneracion"]
+    responses = []
+    for value in data:
+        ts = value.pop("ts")
+        arrow = get(f"{ts} {tz}", "YYYY-MM-DD HH:mm ZZZ")
+        response = {}
+        response["timestamp"] = arrow
+        production = ProductionMix()
+        storage = StorageMix()
+        exchange = {}
+        for mode in value:
```

Comment:
If you still want to have a preprocess function I suggest splitting it out from the fetch data function as they do two very seperate things.